### PR TITLE
fix: Fix for always constant value of transaction.result as success i…

### PR DESCRIPTION
**Pull request by achyut-tokekar**
Fix for always constant value of transaction.result as success in apm

### CODE REVIEW

File: src/instrumentation.ts [hunk 1]
```diff
@@ -246,6 +246,10 @@ export class HyperExpressInstrumentation extends InstrumentationBase {
         const wrapPromise = (promise: Promise<unknown>) => {
           return promise
             .then(value => {
+              if (span._span && span._span._agent && span._span._agent.currentTransaction && span._span._agent.currentTransaction._result && res.statusCode) {
```
Comment (ezioda004) [inline src/instrumentation.ts]: you can use optional chaining

File: src/instrumentation.ts [hunk 2]
```diff
@@ -246,6 +246,10 @@ export class HyperExpressInstrumentation extends InstrumentationBase {
...
+                const statusCodeStartNumber = res.statusCode.toString().charAt(0);
+                span._span._agent.currentTransaction._result = `HTTP ${statusCodeStartNumber}xx`;
```
Comment (ezioda004) [inline src/instrumentation.ts:250]: do we need to set it here? for example _span is not something to be accessed by outside libs

File: src/instrumentation.ts (hunk 1 above)
Comment (achyut-tokekar) [reply src/instrumentation.ts]: done

File: src/instrumentation.ts (hunk 2 above)
Comment (achyut-tokekar) [reply src/instrumentation.ts:250]: added ts-ignore

**Pull request merged**